From b323b439cb991b652494c3b03517e1645f37895c Mon Sep 17 00:00:00 2001
From: Marian Chereji <marian.chereji@freescale.com>
Date: Fri, 11 Oct 2013 17:53:41 +0300
Subject: [PATCH 222/383] dpa_offload: Fix use of bad index in DPA statistics

A bad index was used to update the counter Ids array of the statistics
request in function "get_cnt_us_stats". This was causing severe memory
corruption because the index was overshooting the boundaries of the
request counters Ids array.

Signed-off-by: Marian Chereji <marian.chereji@freescale.com>
Change-Id: Ic781b268ce077e2756d8121b1c836228e38e6b59
Reviewed-on: http://git.am.freescale.net:8181/5506
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Aurelian Zanoschi <Aurelian.Zanoschi@freescale.com>
Reviewed-by: Anca Jeanina Floarea <anca.floarea@freescale.com>
Reviewed-by: Rivera Jose-B46482 <German.Rivera@freescale.com>
[Original patch taken from QorIQ-SDK-V1.5-SOURCE-20131219-yocto.iso]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 drivers/staging/fsl_dpa_offload/dpa_stats.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/dpa_stats.c b/drivers/staging/fsl_dpa_offload/dpa_stats.c
index c5eb276..becbbcb 100644
--- a/drivers/staging/fsl_dpa_offload/dpa_stats.c
+++ b/drivers/staging/fsl_dpa_offload/dpa_stats.c
@@ -3318,7 +3318,7 @@ static int get_cnt_us_stats(struct dpa_stats_req_cb *req_cb,
 			    struct dpa_stats_cnt_cb *cnt_cb)
 {
 	uint32_t i = 0, j = 0;
-	req_cb->config.cnts_ids[cnt_cb->id] = req_cb->bytes_num;
+	req_cb->config.cnts_ids[req_cb->cnts_num] = req_cb->bytes_num;
 
 	for (i = 0; i < cnt_cb->members_num; i++) {
 		for (j = 0; j < cnt_cb->info.stats_num; j++) {
-- 
1.7.5.4

